diff --git a/dist/codegen/lex-gen.js b/dist/codegen/lex-gen.js
index 9bcacea5cdd2f8905eea31948f3f9ad22ab14156..6820baaa053862a3d67f490291766fbc836e7398 100644
--- a/dist/codegen/lex-gen.js
+++ b/dist/codegen/lex-gen.js
@@ -48,7 +48,7 @@ function genCommonImports(file, baseNsid) {
         moduleSpecifier: `${baseNsid
             .split('.')
             .map((_str) => '..')
-            .join('/')}/lexicons`,
+            .join('/')}/lexicons.js`,
     })
         .addNamedImports([{ name: 'validate', alias: '_validate' }]);
     //= import { type $Typed, is$typed as _is$typed, type OmitKey } from '../[...]/util.ts'
@@ -57,7 +57,7 @@ function genCommonImports(file, baseNsid) {
         moduleSpecifier: `${baseNsid
             .split('.')
             .map((_str) => '..')
-            .join('/')}/util`,
+            .join('/')}/util.js`,
     })
         .addNamedImports([
         { name: '$Typed', isTypeOnly: true },
diff --git a/src/codegen/lex-gen.ts b/src/codegen/lex-gen.ts
index 6798b436536c6d5c60b963dac99a737bc1c88b7c..02d298840b61708f57f51050c8b079e063a1f40d 100644
--- a/src/codegen/lex-gen.ts
+++ b/src/codegen/lex-gen.ts
@@ -50,7 +50,7 @@ export function genCommonImports(file: SourceFile, baseNsid: string) {
       moduleSpecifier: `${baseNsid
         .split('.')
         .map((_str) => '..')
-        .join('/')}/lexicons`,
+        .join('/')}/lexicons.js`,
     })
     .addNamedImports([{ name: 'validate', alias: '_validate' }])
 
@@ -60,7 +60,7 @@ export function genCommonImports(file: SourceFile, baseNsid: string) {
       moduleSpecifier: `${baseNsid
         .split('.')
         .map((_str) => '..')
-        .join('/')}/util`,
+        .join('/')}/util.js`,
     })
     .addNamedImports([
       { name: '$Typed', isTypeOnly: true },
